### ISSUE TITLE
Review constrain scikit-learn< 0.23

### ISSUE THREAD

**Issue by holgern** [optuna/optuna#1785]
scikit-optimize 0.8.1 has been released which works together with scikit-learn >= 0.23

https://github.com/scikit-optimize/scikit-optimize/releases/tag/v0.8.1

**Comment by toshihikoyanase** [comment]
@holgern I'm sorry for the delayed response.

> works together with scikit-learn >= 0.23

Thank you for letting us know. The version restriction was introduced by #1236 to resolve #1235, but it is no longer required as you mentioned.

The change seems to be obvious, and let add `contribution-welcome` label to this issue.

## Task

Please remove the version constraint of `scikit-learn<0.23.0` in [`setup.py`](https://github.com/optuna/optuna/blob/master/setup.py).

**Comment by nzw0301** [comment]
We might need to fix codes for `scikit-learn>=0.23.0`. I remove the constraint of `scikit-learn<0.23.0` in `setup.py`, then ran tests:

```bash
$ circleci build --job tests-python37

=========================== short test summary info ============================
FAILED tests/integration_tests/test_sklearn.py::test_objective_error_score_nan
FAILED tests/samplers_tests/test_samplers.py::test_nan_objective_value[<lambda>2]
===== 2 failed, 1189 passed, 3 skipped, 935 warnings in 405.94s (0:06:45) ======
Error:
Exited with code exit status 1

Step failed
Error: runner failed (exited with 101)
Task failed
Step canceled
Error: task failed
```

**Comment by toshihikoyanase** [comment]
@nzw0301 Thank you for your investigation. As you mentioned in #1890, `SkoptSampler` with `scikit-learn>=0.23.0` did not work as expected, and I removed the `contribution-welcome` label.

**Comment by nzw0301** [comment]
It works now...?

```bash
 pytest tests/integration_tests/test_skopt.py            [18:52:05]
====================================== test session starts ======================================
platform darwin -- Python 3.8.6, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /Users/nzw/PycharmProjects/optuna
plugins: anyio-2.1.0
collected 8 items

tests/integration_tests/test_skopt.py ........                                            [100%]

======================================= warnings summary ========================================
tests/integration_tests/test_skopt.py::test_get_trials
  /Users/nzw/PycharmProjects/optuna/optuna/integration/skopt.py:125: ExperimentalWarning: `consider_pruned_trials` option is an experimental feature. The interface can change in the future.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================= 8 passed, 1 warning in 9.62s ==================================
```

```bash
$ pip list | grep -i scikit
scikit-image                  0.18.1
scikit-learn                  0.24.1
scikit-optimize               0.8.1
```

**Comment by nzw0301** [comment]
Ok, not yet :(

**Comment by nzw0301** [comment]
Several PRs of scikit-learn are trying to fix the source of this issue (`0` division of `std`). We will come back to this issue if the PRs have been merged into a future stable version.

**Comment by nzw0301** [comment]
Scikit-learn 0.24.2 is out. It seems to contain the bug fixes related to Gaussian process according to [the release note].(https://scikit-learn.org/stable/whats_new/v0.24.html#sklearn-gaussian-process)

I locally confirmed that the pytest's errors disappear by using this following libraries:

```zsh
scikit-learn                  0.24.2
scikit-optimize               0.8.1
```
